src/main/java/io/codelex/flightplanner/flights/FlightServerInMemory.java

```diff
@@ -0,0 +1,124 @@
+package io.codelex.flightplanner.flights;
+
+import io.codelex.flightplanner.flights.domain.Airport;
+import io.codelex.flightplanner.flights.domain.Flight;
+import io.codelex.flightplanner.flights.domain.SearchFlight;
+import io.codelex.flightplanner.flights.domain.SearchFlightResult;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Locale;
+
+@Service
+@ConditionalOnProperty(prefix = "myapp", name = "appmode", havingValue = "inmemory")
+public class FlightServerInMemory implements FlightService {
+    private FlightRepositoryInMemory repository;
+
+    public FlightServerInMemory(FlightRepositoryInMemory repository) {
+        this.repository = repository;
+    }
+
+    @Override
+    public Flight addFlights(Flight flight) {
+        checkIfSame(flight);
+        checkIfNull(flight);
+        checkSameAirport(flight);
+        checkDate(flight);
+        return repository.addFlights(flight);
+    }
+
+    public void checkIfSame(Flight flight) {
+        List<Flight> sameFlight = repository.flights.stream().filter(flight1 ->
+                flight1.getFrom().equals(flight.getFrom()) &&
+                        flight1.getTo().equals(flight.getTo()) &&
+                        flight1.getCarrier().equals(flight.getCarrier()) &&
+                        flight1.getArrivalTime().equals(flight.getArrivalTime()) &&
+                        flight1.getDepartureTime().equals(flight.getDepartureTime())
+        ).toList();
+        if (sameFlight.size() > 0) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+    }
+
+    public void checkIfNull(Flight flight) {
```

Comment:
Par šo apskaties Spring prezentācijā slaidus par validāciju. Ar pāris anotācijām var izvairīties no šitā garā koda.